astropy/table/bst.py

```diff
@@ -0,0 +1,614 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import operator
+import numpy as np
+from ..extern.six.moves import zip
+
+
+class MaxValue(object):
+    '''
+    Represents an infinite value for purposes
+    of tuple comparison.
+    '''
+    def __gt__(self, other):
+        return True
+
+    def __ge__(self, other):
+        return True
+
+    def __lt__(self, other):
+        return False
+
+    def __le__(self, other):
+        return False
+
+    def __repr__(self):
+        return "MAX"
+
+    __le__ = __lt__
+    __ge__ = __gt__
+    __str__ = __repr__
+
+
+class MinValue(object):
+    '''
+    The opposite of MaxValue, i.e. a representation of
+    negative infinity.
+    '''
+    def __lt__(self, other):
+        return True
+
+    def __le__(self, other):
+        return True
+
+    def __gt__(self, other):
+        return False
+
+    def __ge__(self, other):
+        return False
+
+    def __repr__(self):
+        return "MIN"
+
+    __le__ = __lt__
+    __ge__ = __gt__
+    __str__ = __repr__
+
+
+class Epsilon(object):
+    '''
+    Represents the "next largest" version of a given value,
+    so that for all valid comparisons we have
+    x < y < Epsilon(y) < z whenever x < y < z and x, z are
+    not Epsilon objects.
+
+    Parameters
+    ----------
+    val : object
+        Original value
+    '''
+
```

Comment:
I don't know how common these objects are, but it might be worth having a `__slots__ = ('val',)` for this class.